### DISCUSSION TITLE
How to invoke a function on tour done?

### DISCUSSION THREAD

**Discussion by vegeta123254** [elrumordelaluz/reactour#411] [Q&A]
I am using reactour, v2 .
I have placed TourProvider at App level. 
And at each component , I have used **setSteps** to set the steps. 
Is there a way to invoke a function after the last step is done at each component? (This is basically to make a HTTP request )
I am showing the tours as soon as the components is loaded.

The **beforeClose** method on _TourProvider_ won't work as I have several instances of react tour. 
The **action** method on _steps_ won't work as it is triggered as soon as the step is encountered.

**Comment by elrumordelaluz** [comment]
Hi @vegeta123254, could you make a minimal reproduction of your use case in a codesandbox, in order to play around and look for a solution that fits? thanks!

**Reply by vegeta123254** [reply]
@elrumordelaluz , Please find the [sandbox](https://codesandbox.io/s/angry-satoshi-0h3l7), 
I need to invoke a function when the tour is completed to make an html request.
Thank you in advance.

**Comment by elrumordelaluz** [comment]
[Here](https://codesandbox.io/s/busy-poincare-qfpdb?file=/src/components/C.tsx) is an example making a `fetch` call in two different contexts: one called before set `steps` and opent the _Tour_, using the same `useEffect` hook you used to; the other is inside the `content` Component, as soon as is mounted.

Please let me know if this is what you are looking for.
